packages/framework-core/src/decorators/non-exposed.ts

```diff
@@ -0,0 +1,29 @@
+import { Booster } from '../booster'
+import { AnyClass } from '@boostercloud/framework-types'
+import { getFunctionArguments } from './metadata'
+
+export function NonExposed(
+  target: AnyClass | InstanceType<AnyClass>,
+  methodName: string | undefined,
+  parameterIndex?: number
+) {
+  Booster.configureCurrentEnv((config): void => {
+    const className = target.name ?? target.constructor.name
+    const value: Array<string> = config.nonExposedGraphQLMetadataKey[className] || []
+
+    const fieldName = getFieldName(methodName, target, parameterIndex)
+    config.nonExposedGraphQLMetadataKey[className] = [...value, fieldName]
+  })
+}
+
+function getFieldName(
+  methodName: string | undefined,
+  target: AnyClass | InstanceType<AnyClass>,
+  parameterIndex: number | undefined
+): string {
+  if (methodName) {
+    return methodName
+  }
+  const argumentNames = getFunctionArguments(target)
+  return argumentNames[parameterIndex!]
```

Comment:
I think using `!` will throw if the value is undefined. I know that, supposedly, if `methodName` is undefined, then `parameterIndex` must not... but we can't be sure. My suggestion is to add a explicit throw here with a message that tells exactly that "we could not get field name information in {this class}, {this method name}, etc.". 

That way, we would know instantly what the error is about, instead of needing to debug something like `undefined can be used to index` or, worse, fail silently and having an array of "non exposed fields" all filled with undefined 😂